src/main/java/hudson/cli/CLICommandInvoker.java

```diff
@@ -139,13 +143,32 @@ public Result invoke() {
         final ByteArrayOutputStream out = new ByteArrayOutputStream();
         final ByteArrayOutputStream err = new ByteArrayOutputStream();
 
-        final int returnCode = command.main(
-                args, locale, stdin, new PrintStream(out), new PrintStream(err)
-        );
+        final Charset outCharset;
+        final Charset errCharset;
+        try {
+            outCharset = errCharset = command.getClientCharset();
```

Comment:
This is just going to get the charset of the system running the test. Anyway I do not think that is what we want for tests; use UTF-8.

Reply:
Jesse, when you make requests of me, can you please explain the reasoning? It is very tiresome to accommodate arbitrary requests without understanding why.

Reply:
Or if you do not care, fine, just add a `@SuppressFBWarnings` and leave the impl alone.

Reply:
Well, the production code in `jenkins/core/src/main/java/hudson/cli` uses `getClientCharset()` (i.e. `Charset.defaultCharset()`), so I think it makes sense for our tests to behave the same way. I think our Linux tests should behave the way the Linux production code behaves, and I think our Windows tests should behave the way the Windows production code behaves. This code accomplishes that objective.

Reply:
In a real system the CLI is running on a different machine, and the client charset is intended to come from there. `CLICommandInvoker` is not pretending to simulate that environment. There are tests which do _not_ use this utility and directly invoke `java -jar jenkins-cli.jar`, though currently none of those run the client on another “machine” (Docker container) either.

Reply:
I think there is value in exercising both the UTF-8 code path and the CP-1252 code path in our tests. Sure, the mechanism by which it is exercised might not exactly simulate production, but neither does the mechanism by which `JenkinsRule` loads classes exactly simulate production. In both cases, I think we are making an acceptable tradeoff to get increased test coverage at minimal cost.